src/sources/kubernetes/mod.rs

```diff
@@ -28,9 +30,24 @@ use string_cache::DefaultAtom as Atom;
 /// Location in which by Kubernetes CRI, container runtimes are to store logs.
 const LOG_DIRECTORY: &str = r"/var/log/pods/";
 
+#[derive(Debug, Snafu)]
+enum BuildError {
+    #[snafu(display("To large UID: {:?}", uid))]
+    UidToLarge { uid: String },
+    #[snafu(display("UID contains illegal characters: {:?}", uid))]
+    IllegalCharacterInUid { uid: String },
+}
+
 #[derive(Deserialize, Serialize, Debug, Clone, Default)]
 #[serde(deny_unknown_fields)]
-pub struct KubernetesConfig {}
+pub struct KubernetesConfig {
+    #[serde(default)]
```

Comment:
I believe it should handle it without these defaults? I don't remember needing this for other lists but i could be wrong!

Reply:
Without it Vector complains, but you are also right, they achieved the same having `#[serde(default)]` on struct level. I'll change it to that.